appinventor/blocklyeditor/src/backpack.js

```diff
@@ -397,15 +416,15 @@ Blockly.Backpack.prototype.openBackpack = function() {
   if (!this.isAdded && this.flyout_.isVisible()) {
     this.flyout_.hide();
   } else {
-    var backpack = this.getContents();
-    //get backpack contents from java
-
-    var len = backpack.length;
-    var newBackpack = [];
-    for (var i = 0; i < len; i++) {
-      newBackpack[i] = Blockly.Xml.textToDom(backpack[i]).firstChild;
-    }
-    this.flyout_.show(newBackpack);
+    var p = this;
+    this.getContents(function(backpack) {
```

Comment:
If the backpack is open and the Add All to Backpack option is used, this will result in O(topLevel) getContents calls. I think we should update the condition in line 355 to include a check whether `store` is true and only redraw the backpack if it is being stored since we are otherwise in the middle of an update operation.

Reply:
Another option might be to move lines 354-359 into `setContents` itself since we only want to update the flyout iff its contents have changed.

Reply:
Fix is in the next commit.